process/block/poolscleaner/poolscleaner_test.go

```diff
@@ -0,0 +1,256 @@
+package poolscleaner_test
+
+import (
+	"math/big"
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/ElrondNetwork/elrond-go/data/smartContractResult"
+	"github.com/ElrondNetwork/elrond-go/data/state"
+	"github.com/ElrondNetwork/elrond-go/data/state/addressConverters"
+	"github.com/ElrondNetwork/elrond-go/data/transaction"
+	"github.com/ElrondNetwork/elrond-go/dataRetriever"
+	"github.com/ElrondNetwork/elrond-go/process"
+	"github.com/ElrondNetwork/elrond-go/process/block/poolscleaner"
+	"github.com/ElrondNetwork/elrond-go/process/mock"
+	"github.com/ElrondNetwork/elrond-go/storage"
+	"github.com/stretchr/testify/assert"
+)
+
+func getAccAdapter(nonce uint64, balance *big.Int) *mock.AccountsStub {
+	accDB := &mock.AccountsStub{}
+	accDB.GetExistingAccountCalled = func(addressContainer state.AddressContainer) (handler state.AccountHandler, e error) {
+		return &state.Account{Nonce: nonce, Balance: balance}, nil
+	}
+
+	return accDB
+}
+
+func initDataPoolTransactions() *mock.PoolsHolderStub {
+	return &mock.PoolsHolderStub{
+		TransactionsCalled: func() dataRetriever.ShardedDataCacherNotifier {
+			return &mock.ShardedDataStub{
+				RegisterHandlerCalled: func(i func(key []byte)) {},
+				ShardDataStoreCalled: func(id string) (c storage.Cacher) {
+					return &mock.CacherStub{
+						PeekCalled: func(key []byte) (value interface{}, ok bool) {
+							switch string(key) {
+							case "key1":
```

Comment:
Might change the keys value (also making them consts in this test is a good idea) key1 -> delayedFetchingKey, key2 -> validTxKey, key3 -> invalidTxKey

Reply:
Done.